### DISCUSSION TITLE
Doris-manager在配置SSH信任生成的sshKeyFile，在尝试ssh连接失败

### DISCUSSION THREAD

**Discussion by boneHCL** [apache/doris#9148] [A - General / Q&A]
通过Doris-manager建立集群托管，在配置SSH信任的时候，通过创建manager的sshKeyFile，尝试ssh连接失败。
![image](https://user-images.githubusercontent.com/76100619/164368127-335dd597-4d80-4b74-b458-501e2ee42fe5.png)

异常提示信息如下：
![image](https://user-images.githubusercontent.com/76100619/164368468-e8df5582-7b7f-46d0-9054-dcd09c9a8053.png)

通过日志发现在**SSH.writeSshKeyFile(String sshKey, File sshKeyFile)**
的时候建立的文件格式不符合要求，导致**ResourceNodeAndAgentManager.checkSshConnect()**
命令执行失败：**ssh -o ConnectTimeOut=20 -o StrictHostKeyChecking=no -o BatchMode=yes -o PasswordAuthentication=no -i /home/dev/doris-manager/incubator-doris-manager-1.0.0-rc02/output/server/conf/sshkey-5 -p 22 dev@192.168.2.57 echo ok**

![image](https://user-images.githubusercontent.com/76100619/164369466-daca6f21-861f-4fdf-a6e2-fedf939eb821.png)

和ssh-keygen -t rsa生成的id_rsa对比如下：
![image](https://user-images.githubusercontent.com/76100619/164369257-bd0d2644-293a-485b-b7aa-71d5fa87d404.png)

**Comment by JNSimba** [comment]
1.Did you see that the private key you filled in on the page is exactly the same as the content in the /home/dev/.ssh/id_rsa file?
2.Make sure that the ssh login-free establishment is successful, which can be verified by ssh dev@host on the local machine

**Reply by boneHCL** [reply]
1、确定页面功能中填写的私钥和/home/dev/.ssh/id_rsa文件中的内容一样

![image](https://user-images.githubusercontent.com/76100619/164420936-edf7a4a2-cd26-4410-b7af-c7ee4fc41523.png)

![image](https://user-images.githubusercontent.com/76100619/164421059-8ae23020-5ae2-47e6-b3ed-9b805c918e16.png)


2、我直接使用命令ssh -o ConnectTimeOut=20 -o StrictHostKeyChecking=no -o BatchMode=yes -o PasswordAuthentication=no -p 22 dev@192.168.2.57 echo ok  返回ok。  ssh dev@host 同理

![image](https://user-images.githubusercontent.com/76100619/164420384-4f7aa53f-4fd3-45f1-b9d1-ba207cd2152a.png)

**Reply by JNSimba** [reply]
In theory, the header of sshkey should be included, similar to ---begin open...-----

**Reply by JNSimba** [reply]
That is, your `cat sshkey-5` and `cat /home/dev/.ssh/ida_rsa` should be exactly the same

**Reply by boneHCL** [reply]
除了没有-------BEGIN OPENSSH    和  ------  END  OPENSSH 这个两个文件内容一摸一样。
但是上边的截图显示了，file /home/dev/.ssh/ida_rsa 和 file sshkey-5 结果不一样。我怀疑是不是这个原因导致的。

**Reply by JNSimba** [reply]
I suspect that it is because of the lack of header and footer of the key. You can delete the header and footer in ~/.ssh/ida_rsa and see if ssh can still be successful?

**Reply by boneHCL** [reply]
1. 把 ~/.ssh/ida_rsa 中的页眉和页脚去掉，确实 ssh dev@host 失败，提示：invalid  format
2. 但是我将页面中保存的密钥重新保存，加入ida_rsa中的页眉和页脚，还是失败，原因不变。
3. 我暂时的处理办法一（采用）：改变源码SSH.java中的writeSshKeyFile方法，文件不存在时才创建，第一次异常后，我将ida_rsa 复制并重命名到conf/ 目录下。 办法二（思考）：直接修改writeSshKeyFile将服务器的~/.ssh/ids_rsa文件复制重命名想要的文件。
4. 采用上边的方法，我暂时可以成功创建空间（托管集群）了。
5. 我想说的原因是：程序生成的sshkey-* 文件，通过file sshkey-5 查看是ASKII text，而~/.ssh/ida_rsa 文件的用file命令后，是OpenSSH Private key


感谢，你的提示和回答。

**Reply by JNSimba** [reply]
> 我想说的原因是：程序生成的sshkey-* 文件，通过file sshkey-5 查看是ASKII text，而~/.ssh/ida_rsa 文件的用file命令后，是OpenSSH Private key

I suspect it's the key, I try `file id_rsa` and `file server/conf/sshkey-1` both are the same.
<img width="1060" alt="image" src="https://user-images.githubusercontent.com/7951521/164620576-16be253e-fdf8-4200-940a-c279d8abc6cf.png">

**Reply by boneHCL** [reply]
比较难以理解，我下载doris-manager的源码版本是incubator-doris-manager-1.0.0-rc02，为什么我编译后的系统程序生成的文件格式不对？
